#!/usr/bin/env python
PACKAGE = "franka_example_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("is_scaning",bool_t,0,"a bool parameter indicate if need ext_force",False)
gen.add("translational_stiffness", double_t, 0, "Cartesian translational stiffness, except the Z force", 300, 0, 2000)
gen.add("rotational_stiffness", double_t, 0, "Cartesian rotational stiffness", 50, 0, 200)
gen.add("ext_force",double_t,0,"The contact Force",0,0,10)
gen.add("force_p",double_t,0,"P of PID force controller",1,0,3)
gen.add("force_I",double_t,0,"I of PID force controller",0.1,0,0.5)
gen.add("force_D",double_t,0,"D of PID force controller",0,0,10)

exit(gen.generate(PACKAGE, "dynamic_compliance", "position_force_param"))
